src/consensus/encode.rs

```diff
@@ -306,6 +306,12 @@ pub trait Encodable {
     ///
     /// The only errors returned are errors propagated from the writer.
     fn consensus_encode<W: io::Write + ?Sized>(&self, writer: &mut W) -> Result<usize, io::Error>;
+
+    /// Returns the bytes needed when `self` is serialized
+    fn serialized_len(&self) -> usize {
+        self.consensus_encode(&mut sink()).unwrap()
```

Comment:
In 45df58f08348adb47619687c7b4b7aaf9c863da7 

The name is slightly misleading if this is linear time. If there are types where encoding requires computing some cryptographic operations, we are effectively doing the operation twice.

Even if we don't have types like this before, this is something we have to be mindful of when we implement it for new types in the future.

Reply:
Is the default implementation really necessary?

Reply:
Are you referring only to `fn serialized_len`? No, it's not.

(BTW I am not aware of types that are doing cryptographic operations in encode though)